pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -7,22 +7,35 @@ import (
 	"strings"
 	"time"
 
-	"github.com/grafana/grafana/pkg/services/folder"
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
 	"github.com/grafana/grafana/pkg/services/ngalert/store"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
+type RuleLimits struct {
+	// The default interval if not specified.
+	DefaultRuleEvaluationInterval time.Duration
+	// All intervals must be an integer multiple of this duration.
+	BaseInterval time.Duration
+}
```

Comment:
nit: not a fan of these "partial" [parameter structs](https://refactoring.guru/introduce-parameter-object) I'm all in for all or none. I'd rather see you use the primitives directly e.g. just use `time.Duration` twice

Reply:
i nearly did this, but i also intend to create a generic Limits object in the future. it'll contain all limits... min interval, default interval, rule group size, etc.

i imagine this struct will be combined into this limits object in the future, since the values here kind of act like limits. i chose this pattern for now with this in mind.